### PR TITLE
elixir: Add runnable tasks

### PR DESCRIPTION
**Pull request by jotaviobiondo**
Release Notes:

- Added runnable tasks to Elixir tests (modules, `describe` and `test` blocks)

https://github.com/zed-industries/zed/assets/7737375/06f1b4cb-0364-4c30-982d-6dda3193b5d2

### CODE REVIEW

**Comment by jotaviobiondo** [comment]
@cla-bot check

**Comment by osiewicz** [comment]
Slight nit: could we rename all captures (except for `@run`) to start with the underscore? 
Captures without an underscore prefix are exposed as environmental variable in a task (with `ZED_CUSTOM_` prefix), which means that the user could retrieve full text of `elixir_test` capture by referencing `ZED_CUSTOM_elixir_test` env variable in their task, which I suppose is not desirable.

File: extensions/elixir/languages/elixir/runnables.scm [hunk 1]
```diff
@@ -0,0 +1,20 @@
+; Macros `describe` and `test`.
+; This matches the ExUnit test style.
+(
+    (call
+        target: (identifier) @run (#any-of? @run "describe" "test")
+    ) @elixir-test
```
Comment (osiewicz) [inline extensions/elixir/languages/elixir/runnables.scm:6]:
```suggestion
    ) @_elixir-test
```

File: extensions/elixir/languages/elixir/runnables.scm [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+; Modules containing at least one `describe` or `test`.
+; This matches the ExUnit test style.
+(
+    (call
+        target: (identifier) @run (#eq? @run "defmodule")
+        (do_block
+            (call target: (identifier) @keyword (#any-of? @keyword "describe" "test"))
+        )
+    ) @elixir-module-test
```
Comment (osiewicz) [inline extensions/elixir/languages/elixir/runnables.scm:18]:
```suggestion
            (call target: (identifier) @_keyword (#any-of? @_keyword "describe" "test"))
        )
    ) @_elixir-module-test
```

**Review by osiewicz — CHANGES_REQUESTED** [review]
Slight nit: could we rename all captures (except for `@run`) to start with the underscore? 
Captures without an underscore prefix are exposed as environmental variable in a task (with `ZED_CUSTOM_` prefix), which means that the user could retrieve full text of `elixir_test` capture by referencing `ZED_CUSTOM_elixir_test` env variable in their task, which I suppose is not desirable.

Other than that, this looks good to me!

**Comment by jotaviobiondo** [comment]
Done @osiewicz!

Didn't know about this behavior of captures in Zed, but it's a very nice feature! :)

**Review by osiewicz — APPROVED** [review]
LGTM

**Comment by baldwindavid** [comment]
Thanks for adding @jotaviobiondo ! Would be super cool if `describe` and `test` showed up in the outline (`outline::Toggle`) as well; especially if the outline panel in https://github.com/zed-industries/zed/pull/12637 eventually goes in.

**Pull request merged**